new/reclassify_amazon_linux_platform_family.md

```diff
@@ -0,0 +1,30 @@
+---
+RFC: unassigned
+Author: Sean OMeara <sean@chef.io>
+Status: Draft
+Type: Standards
+---
+
+# Title
+
+Reclassify Amazon Linux platform_family
+
+## Motivation
+
+   As time passes, Amazon Linux looks less and less like a RHEL clone.
+   Release version numbers do not match EL5, EL6, EL7.
+   Amazon is now a rolling release
+   Different package sets are available
+   Package names and contents are different
+   Service providers are different
+
+## Specification
+
+Change Amazon Linux platform_family from 'rhel' to 'amazon'
```

Comment:
So I think we should change the platform_family here to 'fedora' rather than 'amazon'

Having platform_family and platform of fedora/fedora and amazon/amazon doesn't allow leveraging any existing code, and we have to update all the if cases.  If we reuse 'fedora' then code like this just works:

```ruby
case platform_family
when 'rhel', 'fedora'
  # amazon also works fine here
end
```

For code that needs to specifically look at fedora as a distro and cares about things like platform_version then this code will still work:

```ruby
case platform
when 'fedora'
  # inappropriate to match amazon here
  platform_family >= 21 ? "systemd" : "sysvinit"
end
```

There might be some (bad) code which matches on platform_family of 'fedora' and then does very fedora-specific code which would break on 'amazon', but that code should be changed to use platform of 'fedora' instead of 'platform_family'.

This gets us back to the original design goal of platform_family which is to let you be lazy with case statements and have good branching logic.  To that goal it makes more sense to only have the two 'family' buckets where 'rhel' is explicitly 'rebranded redhat enterprise linux with compatible platform_version numbers' and 'fedora' is 'all other fedora-derived distributions'.